include/onnxruntime/core/session/onnxruntime_ep_c_api.h

```diff
@@ -308,6 +308,33 @@ struct OrtKernelImpl {
    * \since Version 1.24.
    */
   ORT_API_T(void, Release, _In_ OrtKernelImpl* this_ptr);
+
+  /** \brief Optional function to pack an initialized constant tensor to the kernel's preferred data layout.
+   *
+   * For example, a Conv kernel can define this function to pack input W to the channel-last data layout.
+   *
+   * \note The kernel is responsible for storing/owning the packed data and related metadata if it sets `is_packed`
```

Comment:
Sharing is big thing on CPU especially between the sessions. Intra session sharing was introduced a little later. So it would make sense to allow it.

Reply:
I now added APIs to allow sharing of pre-packed weights.